app/src/main/java/org/greenstand/android/TreeTracker/models/user/Users.kt

```diff
@@ -36,6 +38,14 @@ class Users(
         return createUser(dao.getUserByWallet(wallet))
     }
 
+    suspend fun checkForUnreadMessagesPerUser(wallet: String): Boolean{
+        return messagesDao.getUnreadMessageCountForWallet(wallet) >= 1
+    }
```

Comment:
You're duplicating the logic that's already in the messages repo. Just use the messagesRepo.

Adding the messagesrepo to user will give a circular dependency error. To solve this, make a new class that syncs messages and move the sync message method from the messagesRepo into that.

Reply:
This is no longer used in the message repo I forgot to delete it in the previous commit.  
Going by the solution you suggested I would just be refactoring the sync logic in the message repo to another class it still caused the same circular dependency error.